l10n_it_fatturapa_out/tests/test_fatturapa_xml_validation.py

```diff
@@ -873,6 +875,55 @@ def test_no_tax_fail(self):
         error_message = f"Invoice {invoice.name} contains product lines w/o taxes"
         self.assertEqual(ue.exception.args[0], error_message)
 
+    def test_partner_no_address_fail(self):
+        """
+        - create an XML invoice where the customer has no address or city
+
+        expect to fail with a proper message
+        """
+        invoice = self._create_invoice()
+        invoice.partner_id.street = False
+        invoice.partner_id.city = False
+        invoice._post()
+        wizard = self.wizard_model.create({})
+        with self.assertRaises(UserError) as ue:
+            wizard.with_context(**{"active_ids": [invoice.id]}).exportFatturaPA()
+        error_msg = ue.exception.args[0]
+        error_fragments = (
+            f"Error processing invoice(s) {invoice.name}",
+            "Indirizzo",
+            "Comune",
+            "Activate debug mode to see the full error",
+        )
+        for fragment in error_fragments:
+            self.assertIn(fragment, error_msg)
+
+        try:
+            # Enter debug mode and add details
+            mock_request = Mock(
+                db=self.env.cr.dbname,
+                env=self.env,
+                website=False,  # compatibility with website module
+                is_frontend=False,
+            )
+            mock_request.session.debug = "assets"
+            odoo.http._request_stack.push(mock_request)
```

Comment:
@aleuffre mi sono messo come co-autore del secondo commit per questa modifica, senza viene sollevato l'errore:

<details>
<summary>Stack</summary>
<pre>
  File "/path/to/l10n-italy/l10n_it_fatturapa_out/tests/test_fatturapa_xml_validation.py", line 913, in test_partner_no_address_fail
    wizard.with_context(**{"active_ids": [invoice.id]}).exportFatturaPA()
  File "/path/to/l10n-italy/l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py", line 282, in exportFatturaPA
    attach = self.saveAttachment(fatturapa, progressivo_invio)
  File "/path/to/l10n-italy/l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py", line 57, in saveAttachment
    attach_str = fatturapa.to_xml(self.env)
  File "/path/to/l10n-italy/l10n_it_fatturapa_out/wizard/efattura.py", line 277, in to_xml
    content = ir_ui_view._render_template(
  File "/path/to/odoo/odoo/addons/base/models/ir_ui_view.py", line 2127, in _render_template
    return self.env['ir.qweb']._render(template, values)
  File "/path/to/odoo/odoo/tools/profiler.py", line 292, in _tracked_method_render
    return method_render(self, template, values, **options)
  File "/path/to/odoo/odoo/addons/base/models/ir_qweb.py", line 580, in _render
    irQweb = self.with_context(**options)._prepare_environment(values)
  File "/path/to/odoo/addons/http_routing/models/ir_qweb.py", line 12, in _prepare_environment
    irQweb = super()._prepare_environment(values)
  File "/path/to/odoo/odoo/addons/base/models/ir_qweb.py", line 909, in _prepare_environment
    elif 'disable-t-cache' in debug:
TypeError: argument of type 'Mock' is not iterable
</pre>
</details>

Reply:
@SirAionTech va riportata anche in 14?

Reply:
No, è un adattamento che ho dovuto fare per far funzionare il test in `16.0`

Reply:
va benissimo, ci mancherebbe! Grazie per la notifica.